app/assets/javascripts/discourse/tests/unit/controllers/topic-test.js

```diff
@@ -74,6 +74,29 @@ module("Unit | Controller | topic", function (hooks) {
     assert.ok(destroyed, "destroy not popular topic");
   });
 
+  test("deleteTopic permanentDelete", function (assert) {
+    const model = Topic.create();
+    let destroyed = false;
+    model.destroy = async () => (destroyed = true);
```

Comment:
```suggestion
    const stub = sinon.stub(model, "destroy")
```

and later:

```js
sinon.assert.calledOnce(stub);
```

(and needs `import sinon from "sinon";`)